### ISSUE TITLE
Crash in C4ScriptGuiWindow::ChildWithIDRemoved.

### ISSUE THREAD

**Issue by fohly** [openclonk/openclonk#104]
Windows 64 bit. Debug build. 
(ReleaseWithDebInfo won't have the crash.)

Start "The Guardians of Windmills" in Fullscreen.
Press "P" to bring up the shop menu.
Press "Esc" and quit the scenario.
Crash!

I'm actually not sure whether bringing up the menu is related or not.
But it's what I did when the crash happened and the Debug build is painfully slow, so I didn't try it without yet.

Extra Info (in ChildWithIDRemoved):
The childrenIDMap seems corrupt, it reports `childrenIDMap = { size=15987178197214944733 }`.
The "child" pointer seems to point to uninitialized memory. The whole object is initialized to 0xdd bytes.

But perhaps some of the stack is corrupted, since further up the call stack, the child pointer is null, but it is not null once it reaches ChildWithIDRemoved.

Stacktrace:
```
 	openclonk-debug.exe!std::_Tree<std::_Tmap_traits<int,C4ScriptGuiWindow * __ptr64,std::less<int>,std::allocator<std::pair<int const ,C4ScriptGuiWindow * __ptr64> >,1> >::_Eqrange<int>(const int & _Keyval) Line 1989	C++
 	openclonk-debug.exe!std::_Tree<std::_Tmap_traits<int,C4ScriptGuiWindow * __ptr64,std::less<int>,std::allocator<std::pair<int const ,C4ScriptGuiWindow * __ptr64> >,1> >::_Eqrange<int>(const int & _Keyval) Line 2025	C++
 	openclonk-debug.exe!std::_Tree<std::_Tmap_traits<int,C4ScriptGuiWindow * __ptr64,std::less<int>,std::allocator<std::pair<int const ,C4ScriptGuiWindow * __ptr64> >,1> >::equal_range(const int & _Keyval) Line 1590	C++
>	openclonk-debug.exe!C4ScriptGuiWindow::ChildWithIDRemoved(C4ScriptGuiWindow * child) Line 1231	C++
 	openclonk-debug.exe!C4ScriptGuiWindow::RemoveChild(C4ScriptGuiWindow * child, bool close, bool all) Line 1317	C++
 	openclonk-debug.exe!C4ScriptGuiWindow::ClearChildren(bool close) Line 1327	C++
 	openclonk-debug.exe!C4ScriptGuiWindow::Close() Line 1334	C++
 	openclonk-debug.exe!C4ScriptGuiWindow::ClearPointers(C4Object * pObj) Line 1170	C++
 	openclonk-debug.exe!C4ScriptGuiWindow::ClearPointers(C4Object * pObj) Line 1186	C++
 	openclonk-debug.exe!C4Game::ClearPointers(C4Object * pObj) Line 934	C++
 	openclonk-debug.exe!C4ObjectList::DeleteObjects() Line 996	C++
 	openclonk-debug.exe!C4GameObjects::DeleteObjects(bool delete_inactive_objects) Line 255	C++
 	openclonk-debug.exe!C4Game::DeleteObjects(bool fDeleteInactive) Line 1054	C++
 	openclonk-debug.exe!C4Game::Clear() Line 599	C++
 	openclonk-debug.exe!C4Application::GameTick() Line 708	C++
 	openclonk-debug.exe!C4ApplicationGameTimer::Execute(int iTimeout, pollfd * __formal) Line 908	C++
 	openclonk-debug.exe!StdScheduler::DoScheduleProcs(int iTimeout) Line 78	C++
 	openclonk-debug.exe!C4AbstractApp::DoScheduleProcs(int iTimeout) Line 75	C++
 	openclonk-debug.exe!StdScheduler::ScheduleProcs(int iTimeout) Line 145	C++
 	openclonk-debug.exe!C4AbstractApp::Run() Line 56	C++
 	openclonk-debug.exe!WinMain(HINSTANCE__ * hInst, HINSTANCE__ * hPrevInstance, char * lpszCmdParam, int nCmdShow) Line 95	C++
 	openclonk-debug.exe!invoke_main() Line 99	C++
 	openclonk-debug.exe!__scrt_common_main_seh() Line 253	C++
 	openclonk-debug.exe!__scrt_common_main() Line 296	C++
 	openclonk-debug.exe!WinMainCRTStartup() Line 17	C++
 	kernel32.dll!BaseThreadInitThunk()	Unknown
 	ntdll.dll!RtlUserThreadStart()	Unknown
```

**Comment by isilkor** [comment]
0xDD marks the contents of a freed block in MSVC's debug heap implementation. @walachey, the script-based GUI code was yours, right? Can you take a look?

**Comment by walachey** [comment]
Yeah, I'll put it on my todo. Thanks for pinging me

**Comment by walachey** [comment]
Closed in e9fecd4b07ec5bd789658c38846a16bf8282e243
Also see dc3dda4f354b7920eb6e2413e6d08dfd30d75cae
